Improve accessibility vol 2

ahukkanen opened the pull request:
#### :tophat: What? Why?

The great work that got started by [MoretS](https://github.com/decidim/decidim/pull/5684) continues.

This pull request is a big one but fixes many accessibility issues in Decidim. More information about the fixed issues is available from:
https://trello.com/b/gu7Ztm3C/decidim-accessibility

These issues have been reported by accessibility experts for two cities in Finland.

There are still some issues remaining after this but this addresses all the "extremely important" and "important" level issues, which should be enough to pass most accessibility reviews.

A thing to note is also that the accessibility reviews have been done on customized Decidim themes, so the color contrasts of the default theme have not been reviewed by any experts. That is left for further evaluation.

#### :pushpin: Related Issues
- Related to #5684, #5942
- Fixes
  * https://meta.decidim.org/processes/bug-report/f/210/proposals/15108
  * https://meta.decidim.org/processes/bug-report/f/210/proposals/15109
  * https://meta.decidim.org/processes/bug-report/f/210/proposals/15112

#### :clipboard: Subtasks
- [ ] Add `CHANGELOG` entry
- [x] Add tests

andreslucena commented:
Hi @ahukkanen 
Wow, a titanic work :clap: :clap: 
Is it ready to review or it will be finished when there's nothing on the backlog column on Trello?

ahukkanen commented:
@andreslucena Yes, it is ready for review. Those issues in the Trello backlog will remain for the next round, when ever that is... :)

I moved that one issue marked "in progress" back to "backlog" until I will get further information about it. One of the reported issues regarding that has been addressed but I've got another report about that same issue which remains unclear. The Helsinki reviewers said there should be "lang" attribute also in the search field but I haven't experienced such issue they reported with any of the screen readers I've tested myself, so I won't touch it before it is clarified.

The accessibility experts said that those issues marked with criticality levels 1-2 are just "nice to have". Although the heading issue is marked with a higher priority in the Helsinki review but I don't know a proper way to solve it right now to be honest (it's been improved slightly but still remains an issue, I wrote about it in the email).

I may still be updating small pieces here and there, I've been mostly trying to keep it in sync with `develop`. One thing on my list is another round of HTML validation (since there's been a lot of changes since the previous round) but that shouldn't affect this that much as a whole from the reviewer's perspective.

I'd be more than happy this got merged rather sooner than later. Further issues can be handled with their own PRs.

ahukkanen commented:
The second round of HTML validation is now done. I've gone through most of the other views but I've skipped the initiatives module because it's kind of irrelevant in Finland. There is also a couple of HTML validator warnings (empty heading elements) in the single consultation view I didn't touch because they were only warnings.

Now I think it's just waiting for the reviewer comments and trying to keep it in sync with `develop`.

Note that the assemblies specs are currently failing due to the randomly failing `edit_link_shared_examples.rb` specs. Just re-run the tests (possibly a couple of times) and it should be fine.

ivan-mr commented:
Please @ahukkanen , can you fix the Assemblies test?
Thanks!

ahukkanen commented:
@ivan-mr It's not broken by this PR as I explained above. The test itself is not working correctly.

If you want a fix for that, re-run the job or merge #6161.

andreslucena commented:
Hi all, 
Just for giving transparency on how we're tackling this huge issue: 
First, some background information: on the past we received the PR #5684 by @MoretS for improving Accessibility. Then @ahukkanen also contacted us that he also needed to work with these issues, so we decided to merge what was on this first PR so we could get the ball going. At the moment we (@decidim/product) found some critical UX details that we want to fix before making a release with all these changes, so we're looking for an UX designer that also know about A11Y.
For the next release (v0.22), we're reverting (#6195) the first PR so we don't block future releases with this issue.
We're hoping to have this issues fixed and this PR merged at September / October. 
Thanks everyone for this collective effort, specially for @ahukkanen and @MoretS for your outstanding work.
You can see all the pending issues and their status on the [Accessibility - a11y Project](https://github.com/decidim/decidim/projects/15).

microstudi commented:
Hi everyone,

In @decidim/core we've been trying to follow what @andreslucena has been explained: revert the accessibility changes in the 0.22 release and pospone the fixing of the UX/UI details in the 0.23.

However, it turned out more complicated than expected. The main problem is that many contributions have been merged after the #5684 (this is a very active period of development). Contributions that changed code in the same lines, in some cases non-trivial refactors. So, in order to be able to effectively revert it, many conflicts have to be solved and then additional modifications in the files have to be done to revert posterior commits. It's quite a mess...

So, in the end, we've come to the agreement that it is better not to waist resources reverting those changes (as we will want them back anyway in the next version) but **delay** the release of the 0.22 version of Decidim until we fix the UX/UI issues. We expect between 2/4 weeks until the final release of 0.22 version (it needs a period of staging/test in meta-decidim).

What we will do instead is to advance the UX "fixing" task so the 0.22 will incorporate the accessibility corrections without a major break in the design. 

@ahukkanen as we will dedicate next days/week to work on it, we wanted to review this PR as well. So, unless you tell me no to, we'll start to review this, we might even merge it in order to see all the UX changes at once. However, it is a very long PR and it will depend on how the review process goes.

We are sorry for the inconvenience for delaying the release but given the resources and circumstances we really think it is the best. 
Feel free to suggest or ask anything.
Thanks

ahukkanen commented:
@microstudi By all means, it's ready for review on our side.

microstudi commented:
This PR has been superseded by #6253 into develop and #6323 for the 0.22 branch

tramuntanal commented:
Closing as it has been superseded by #6253
